api/src/main/java/org/openmrs/module/referencedemodata/obs/DemoObsGenerator.java

```diff
@@ -73,22 +86,39 @@ public DemoObsGenerator(DemoDataConceptCache conceptCache) {
 	
 	public void createDemoVitalsObs(Patient patient, Encounter encounter, Location location) {
 		for (NumericObsValueDescriptor vitalsDescriptor : getVitalsDescriptors()) {
-			createNumericObsFromDescriptor(vitalsDescriptor, patient, encounter, location);
+			createNumericObsFromDescriptor(vitalsDescriptor, patient, encounter, encounter.getEncounterDatetime(), location);
 		}
 	}
 	
 	public void createDemoLabObs(Patient patient, Encounter encounter, Location location) {
-		for (NumericObsValueDescriptor labDescriptor : getLabDescriptors()) {
-			createNumericObsFromDescriptor(labDescriptor, patient, encounter, location);
+		List<Concept> labConcepts = new ArrayList<>();
+		// Add panels thrice to increase chance of being picked
+		labConcepts.addAll(conceptCache.getConceptsByClass("LabSet"));
```

Comment:
This is quite wasteful. We're making a binary choice and could just use `shouldRandomEventOccur(.6666)` to chose whether we're generating a single test or a panel and then draw things from there.